scheduler/test/cook/test/mesos/api.clj

```diff
@@ -1216,4 +1220,43 @@
                (api/retrieve-url-path target-framework-id agent-hostname "executor-102")))
         (is (nil? (api/retrieve-url-path target-framework-id agent-hostname "executor-103")))
         (is (= (str "http://" agent-hostname ":5051/files/read.json?path=%2Fpath%2Ffor%2Fexecutor-111")
-               (api/retrieve-url-path target-framework-id agent-hostname "executor-111")))))))
\ No newline at end of file
+               (api/retrieve-url-path target-framework-id agent-hostname "executor-111")))))))
+
+(deftest test-instance-progress
+  (let [uri "datomic:mem://test-instance-progress"
+        conn (restore-fresh-database! uri)
+        driver (reify msched/SchedulerDriver)
+        driver-atom (atom nil)
+        fenzo (sched/make-fenzo-scheduler driver-atom 1500 nil 0.8)
+        make-status-update (fn [task-id reason state progress]
+                             (let [task {:task-id {:value task-id}
+                                         :reason reason
+                                         :state state
+                                         :data (ByteString/copyFrom (.getBytes (pr-str {:percent progress}) "UTF-8"))}]
+                               task))
+        job-id (create-dummy-job conn :user "user" :job-state :job.state/running)
+        send-status-update #(async/<!!
+                              (sched/handle-status-update conn driver fenzo
```

Comment:
In the cook scheduler, the progress will be sent as framework messages. We may need to edit this test for consistency while merging the cook scheduler branch.